src/services/narrowing.ts

```diff
@@ -123,4 +123,19 @@ export function narrowFilterOperator(op: string): LabelFilterOp | NumericFilterO
   }
 }
 
+export function narrowUrlParam(param: unknown): string[] | null {
```

Comment:
Excellent! My only suggestion would be to make it generic so we can reuse this narrowing elsewhere, maybe like:

```suggestion
export function narrowStringsArray(param: unknown): string[] | null {
```

I wanted to make it generic but I couldn't find an easy way.

Reply:
Yeah if there was a generic narrowing strategy it would be the most imported typescript npm package, I think this is something that needs a bit of hand-holding

Reply:
there is a `unknownToStrings` method in narrowing already

Reply:
I tried `unknownToStrings` but it's a slightly different use case where the param are passed in as strings and not an array so it never gets past 
```
if (Array.isArray(a)) {
```

Reply:
Updated the func name.